logic/Rule.java

```diff
@@ -168,25 +168,23 @@ public final int hashCode() {
     }
 
     public void validateSatisfiable() {
-        /*
-         check that none of the variables in the `when` or `then` conjunctions are marked `isSatisfiable = false`. otherwise throw an error
-         */
-    }
+        Stream.concat(then.variables().stream(), when.variables().stream()).forEach(variable -> {
+            if (!variable.isSatisfiable()) throw GraknException.of(RULE_CANNOT_BE_SATISFIED, structure.label(), variable.reference().toString());
+        });
 
-    public void validateInsertable()  {
-        /*
-        High level, we also want to ensure that every combination of possible variable types in the `then`, that may be provided
-        by the `when` of the rule, is actually compatible with the `then` of the rule. This means,
-        we need _any_ set of instances to be insertable in the `then`. Note that inserts are different from `match`
-        in that you don't check any subtyping (i think), just check what the exact concept's capabilities are.
+//        if (Stream.concat(then.variables().stream(), when.variables().stream()).anyMatch(variable -> !variable.isSatisfiable())) {
+//            throw GraknException.of(RULE_CANNOT_BE_SATISFIED, structure.label());
+//        }
+    }
 
-        If any combination of types the `when` may produce is not compatible with the `then`, we should flag it to the user.
+    public void validateInsertable() {
+        ResourceIterator<Map<Reference.Name, Label>> possibleWhenPerms = logicManager.typeResolver().retrievePossibleTypeCombos(when, false);
+        Set<Map<Reference.Name, Label>> possibleThenSet = logicManager.typeResolver().retrievePossibleTypeCombos(then, true).toSet();
 
-        To do this you may want to
-        1. utilise the streaming mode of `TypeResolver` (can re-run it, thats ok) to get all combinations
-        2. utilise the `Rule.Conclusion` classes before (there's exactly 1 of the 3 per rule) to validate a combination is compatible with the conclusion
-           using these Rule.Conclusion data structures indicates that it is a "high level" logical rule validation, which is true
-         */
+        possibleWhenPerms.forEachRemaining(nameLabelMap -> {
+            if (possibleThenSet.stream().noneMatch(thenMap -> nameLabelMap.entrySet().containsAll(thenMap.entrySet())))
```

Comment:
All named vars in the `then` are bound in the `when`, but not the other way around. hence use `containsAll` not `equals`.